libs/community/langchain_community/llms/cleanlab.py

```diff
@@ -0,0 +1,162 @@
+import logging
+from typing import Any, Dict, List, Mapping, Optional
+
+from langchain_core.callbacks import (
+    AsyncCallbackManagerForLLMRun,
+    CallbackManagerForLLMRun,
+)
+from langchain_core.language_models.llms import BaseLLM
+from langchain_core.outputs import Generation, LLMResult
+from langchain_core.utils import convert_to_secret_str, get_from_dict_or_env
+from pydantic import Field, PrivateAttr, SecretStr, model_validator
+
+from langchain_community.llms.utils import enforce_stop_tokens
+
+logger = logging.getLogger(__name__)
+
+
+class TrustworthyLanguageModel(BaseLLM):
```

Comment:
Let's call this CleanlabLLM

as an aside, almost all LLM (str->str) usage has migrated to chat models (messages->message) in the past year if relevant to your integration roadmap

```suggestion
class CleanlabLLM(BaseLLM):
```

Reply:
most of the LLM integrations i referred to base this PR on still used `str` -> `str` interface (taking [anthropic](https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/llms/anthropic.py#L216) example here).

also, current TLM integration returns `LLMResult` object, which again i found commonly used across other LLM integrations (taking anyscale [example here](https://github.com/langchain-ai/langchain/blob/9fedb04dd38c377ad7293cab106b041793c9f998/libs/community/langchain_community/llms/anyscale.py#L251))

i found LLM partner pkgs use `AIMessage` as a return object (example [here](https://github.com/langchain-ai/langchain/blob/9fedb04dd38c377ad7293cab106b041793c9f998/libs/partners/openai/langchain_openai/chat_models/base.py#L680)).

as you rightly figured, TLM internally wraps around openai/anthropic, taking prompt & few configs as input. TLM would return: `str` response, and metadata (`trustworthiness_score`, `explanation`, more in future).

can you suggest if we should still consider migrating to chat model (`messages`->`messages`)?